XAMLTest/Internal/Validation.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Globalization;
+using System.Windows;
+using System.Windows.Controls;
+
+namespace XamlTest.Internal;
+
+internal static class Validation
+{
+    internal static readonly DependencyProperty ErrorProperty = DependencyProperty.RegisterAttached(
+        $"Error-{Guid.NewGuid()}", typeof(object), typeof(Validation), new PropertyMetadata(default(object)));
+
+    internal static void SetValidationErrorDummy(DependencyObject element, object value)

```

Comment:
Consider renaming the set/get methods for the attached property to something a little more indicative. I wasn't very creative when I came up with these names :-) Also, they no longer match the name defined on the static DependencyProperty. I guess that is not an issue here because it is only used internally (and not from XAML).